File: 1354-find-players-with-zero-or-one-losses/find-players-with-zero-or-one-losses.cpp
class Solution {
public:
    vector<vector<int>> findWinners(vector<vector<int>>& matches) {
        unordered_map<int, int> winner;
        unordered_map<int, int> looser;

        for (auto it : matches) {
            winner[it[0]]++;
            looser[it[1]]++;
        }

        vector<int> notlost;
        vector<int> lost1;

        for (auto it : winner) {
            int playerId = it.first;
            if (looser.find(playerId) == looser.end()) {
                notlost.push_back(playerId);
            }
        }

        for (auto it : looser) {
            int playerId = it.first;
            if (it.second == 1) {
                lost1.push_back(playerId);
            }
        }

        sort(notlost.begin(), notlost.end());
        sort(lost1.begin(), lost1.end());

        vector<vector<int>> ans;
        ans.push_back(notlost);
        ans.push_back(lost1);

        return ans;
    }
};

File: 1354-find-players-with-zero-or-one-losses/find-players-with-zero-or-one-losses.java
class Solution {
    public List<List<Integer>> findWinners(int[][] matches) {
        HashMap<Integer,Integer>winner=new HashMap<>();
        HashMap<Integer,Integer>looser=new HashMap<>();
        for(int []match:matches){
            winner.put(match[0],winner.getOrDefault(match[0],0)+1);
            looser.put(match[1],looser.getOrDefault(match[1],0)+1);

        }
        ArrayList<Integer>notLost=new ArrayList<>();
        ArrayList<Integer>lost1=new ArrayList<>();
        for(int playerId:winner.keySet())
        {
            if(!looser.containsKey(playerId)){
                notLost.add(playerId);
            }
        }
        for(int playerId:looser.keySet()){
            if(looser.get(playerId)==1){
                lost1.add(playerId);
            }
        }
        Collections.sort(lost1);
        Collections.sort(notLost);
        ArrayList<List<Integer>>ans=new ArrayList<>();
        ans.add(notLost);
        ans.add(lost1);
        return ans;
    }
}